Rel Notes for Storage 4.22 (#109434)
openshift/openshift-docs: modules/rn-ocp-release-notes-new-features.adoc

Detailed information.
////

[id="ocp-release-notes-storage_{context}"]
== Storage

[id="release-notes-storage-images-snapshot-class_{context}"]

> === Creating VMs from templates saved as snapshots (golden images) is generally available

Reviewer: I wouldn't call it this way. Instead use "New VolumeSnapshotClass csi-gce-pd-vsc-images"

Author: Fixed.

By default, you cannot restore more than six volumes per snapshot per hour. So in Kubevirt environments, you normally cannot create more than six VMs per hour from a "golden image" (templates saved as snapshots).

> For Google Cloud Platform (GCP) persistent disk (PD) storage CSI, there is now a non-default `VolumeSnapshotClass`, `csi-gce-pd-vsc-images`, that includes the `snapshot-type: images` parameter. When using KubeVirt, this images snapshot class allows you overcome the six VMs per hour restriction, so that you can create VMs from "golden images". 

Reviewer 2: This line sounds confusing to me.

Would it be better to re-phrase it a little?
```suggestion
For Google Cloud Platform (GCP) persistent disk (PD) storage CSI, there is now a non-default `VolumeSnapshotClass` (named `csi-gce-pd-vsc-images`) that uses the `snapshot-type: images` parameter. When using KubeVirt, it allows you to overcome the six VMs per hour restriction, so that you can create VMs from "golden images". 
```

Author: Fixed.

This feature is generally available in {product-title} 4.22.

For more information, see xref:../storage/container_storage_interface/persistent-storage-csi-snapshots.adoc#volume-snapshot-crds[Volume snapshots CRD: VolumeSnapshotClass].

[id="release-notes-storage-adding-bm-nodes-vsphere_{context}"]
=== Adding bare-metals nodes on vSphere is generally available
{product-title} 4.21 added the ability to add bare-metal nodes to an {product-title} cluster on vSphere as a Technology Preview feature. In {product-title} 4.22, this feature is generally available.

> However, if you add bare-metal nodes, you must remove the vSphere CSI Driver, otherwise the cluster is marked as degraded. 

Reviewer 3: There should be a link to how to remove the CSI driver, https://docs.redhat.com/en/documentation/openshift_container_platform/4.21/html/storage/using-container-storage-interface-csi#persistent-storage-csi-vsphere-disable-storage-procedure_persistent-storage-csi-vsphere

Author: Added it.

* When you require write access to the same volume in multiple zones

For a single Hyperdisk Balanced High Availability volume, you can specify up to 100,000 IOPS and 1,200 MiB/s of throughput.

For more information, see xref:../storage/container_storage_interface/persistent-storage-csi-gcp-pd.adoc#c3-instance-type-for-bare-metal-and-n4-machine-series[Hyperdisk Balanced High Availability disks overview].

[id="release-notes-storage-lso-symlinks_{context}"]
=== Local Storage Operator symlinks management is generally available
To prevent storage breakage during {product-title} upgrades, {product-title} 4.22 provides a mechanism, the `LocalVolumeDeviceLink` Custom Resource Definition, to detect, alert, and remap broken symlinks without manual node-level intervention.

The Local Storage Operator (LSO) traditionally creates persistent volumes (PVs) based on `/dev/disk/by-id/` paths, following the assumption that they are stable. However, Linux kernel updates, firmware updates, or `udev` rule changes can cause these supposedly stable names to change or disappear.

Administrators have the following notification and correction options to deal with symlink disruptions:

> * Monitoring: (default) If the current and preferred path do not match, an alerts occurs, but no changes occur to the current path.

Reviewer 2: ```suggestion
* Monitoring: (default) If the current and preferred path do not match, an alert occurs, but no changes occur to the current path.
```

Author: Good catch. Thanks!

* Use existing path: Alerts are silenced and LSO uses the existing path.

* Recreate symlinks: Symlinks are recreated to point to the new, updated device path.

For more information, see xref:../storage/persistent_storage_local/persistent-storage-local.adoc#local-storage-operator-symlinks-management[Local Storage Operator symlinks management].

> 

Reviewer: missing Mutable CSI Node Allocatable Property (GA)
https://redhat.atlassian.net/browse/OCPSTRAT-2866

Author: I've added it.

This feature is generally available in {product-title} 4.22.

For more information, see xref:../storage/container_storage_interface/persistent-storage-csi-snapshots.adoc#volume-snapshot-crds[Volume snapshots CRD: VolumeSnapshotClass].

[id="release-notes-storage-adding-bm-nodes-vsphere_{context}"]
=== Adding bare-metals nodes on vSphere is generally available
{product-title} 4.21 added the ability to add bare-metal nodes to an {product-title} cluster on vSphere as a Technology Preview feature. In {product-title} 4.22, this feature is generally available.

However, if you add bare-metal nodes, you must remove the vSphere Container Storage Interface (CSI) Driver, otherwise the cluster is marked as degraded. 

For information about how to add bare-metal nodes, see xref:../storage/container_storage_interface/persistent-storage-csi-vsphere.adoc#persistent-storage-csi-vsphere-adding-bm-nodes_persistent-storage-csi-vsphere[Adding bare-metal nodes].

> For information about how to removed the vSphere CSI Driver, see xref:../storage/container_storage_interface/persistent-storage-csi-vsphere.adoc#persistent-storage-csi-vsphere-disable-storage-overview_persistent-storage-csi-vsphere[Disabling and enabling storage on vSphere].

Reviewer: "How to remove"

Author: Good catch. Thanks!

[id="release-notes-storage-hyperdisk-balanced-ha-support_{context}"]
=== Support for Hyperdisk Balanced High Availability volumes is generally available
{product-title} 4.22 introduces support for Hyperdisk Balanced High Availability volumes as generally available.

Hyperdisk Balanced High Availability volumes are useful for:

* Protecting your applications from a zonal outage by synchronously replicating data across two zones in the same region
* When you require write access to the same volume in multiple zones

> For a single Hyperdisk Balanced High Availability volume, you can specify up to 100,000 IOPS and 1,200 MiB/s of throughput.

Reviewer: why 1200MiB/S? 
https://docs.cloud.google.com/compute/docs/disks/hd-types/hyperdisk-balanced-ha
says 2400

Author: Fixed.

Detailed information.
////

[id="ocp-release-notes-storage_{context}"]
== Storage

[id="release-notes-storage-images-snapshot-class_{context}"]

> === New VolumeSnapshotClass csi-gce-pd-vsc-images is generally available

Reviewer 4: LGTM

By default, you cannot restore more than six volumes per snapshot per hour. So in Kubevirt environments, you normally cannot create more than six VMs per hour from a "golden image" (templates saved as snapshots).

For Google Cloud Platform (GCP) persistent disk (PD) storage Container Storage Interface (CSI), there is now a non-default `VolumeSnapshotClass`, named `csi-gce-pd-vsc-images`, that uses the `snapshot-type: images` parameter. When using KubeVirt, it allows you to overcome the six VMs per hour restriction, so that you can create VMs from "golden images". 

This feature is generally available in {product-title} 4.22.

For more information, see xref:../storage/container_storage_interface/persistent-storage-csi-snapshots.adoc#volume-snapshot-crds[Volume snapshots CRD: VolumeSnapshotClass].

[id="release-notes-storage-adding-bm-nodes-vsphere_{context}"]

> === Adding bare-metals nodes on vSphere is generally available

Reviewer 4: LGTM

However, if you add bare-metal nodes, you must remove the vSphere Container Storage Interface (CSI) Driver, otherwise the cluster is marked as degraded. 

For information about how to add bare-metal nodes, see xref:../storage/container_storage_interface/persistent-storage-csi-vsphere.adoc#persistent-storage-csi-vsphere-adding-bm-nodes_persistent-storage-csi-vsphere[Adding bare-metal nodes].

For information about how to remove the vSphere CSI Driver, see xref:../storage/container_storage_interface/persistent-storage-csi-vsphere.adoc#persistent-storage-csi-vsphere-disable-storage-overview_persistent-storage-csi-vsphere[Disabling and enabling storage on vSphere].

[id="release-notes-storage-hyperdisk-balanced-ha-support_{context}"]
=== Support for Hyperdisk Balanced High Availability volumes is generally available
{product-title} 4.22 introduces support for Hyperdisk Balanced High Availability volumes as generally available.

Hyperdisk Balanced High Availability volumes are useful for:

* Protecting your applications from a zonal outage by synchronously replicating data across two zones in the same region
* When you require write access to the same volume in multiple zones

> For a single Hyperdisk Balanced High Availability volume, you can specify up to 100,000 IOPS and 2,400 MiB/s of throughput.

Reviewer: remove this line

Author: Done.

Author: Done.

* Use existing path: Alerts are silenced and LSO uses the existing path.

* Recreate symlinks: Symlinks are recreated to point to the new, updated device path.

For more information, see xref:../storage/persistent_storage_local/persistent-storage-local.adoc#local-storage-operator-symlinks-management[Local Storage Operator symlinks management].

[id="release-notes-storage-mutable-csi-node-allocatable-property_{context}"]
=== Mutable CSI node allocatable property is generally available
This feature allows for dynamically updating the maximum number of storage volumes a node can handle. Without this feature, volume limits are essentially immutable when a node first joins the cluster. If the environment changes—for example, if you attach a new network interface (ENI) that shares a hardware "slot" with your storage—OpenShift Container Platform does not recognize it has fewer slots available for disks, leading to pods becoming stuck.

This feature is only supported on AWS Elastic Block Storage (EBS).

Mutable CSI node allocatable property was introduced in {product-title} 4.21 as a Technical Preview feature. In {product-title} 4.22, it is supported as generally available.

> To enable this feature, you need to enable Feature Gates.

Reviewer: it's GA so no feature gate

Author: Fixed.

For more information about enabling Feature Gates, see xref:../nodes/clusters/nodes-cluster-enabling-features.adoc[Feature Gates].

[id="release-notes-storage-sovereign-cloud_{context}"]
=== European Sovereign Cloud (EUSC) region (Technology Preview)
European Sovereign Cloud (EUSC) region acts as a "digital fortress" built within a specific country’s borders. Sovereign Clouds are specifically designed to meet strict legal, jurisdictional, and security requirements of a particular nation or entity.

In the context of storage, EUSC ensures that all data, including primary storage, backups, and the resulting metadata, resides physically within the specific nation’s borders and remains exclusively under its legal jurisdiction.

For {product-title} 4.22, only AWS Elastic Block Storage supports EUSC. AWS Elastic File Storage (EFS) is not supported.

EUSC is supported as a Technology Preview feature.

> For more information about EUSC, see ???

Reviewer: need link here

Author: Yes. Will not be able to link until I merge the main content.

Author: Fixed.
